magnum_cluster_api/driver.py

```diff
@@ -323,7 +323,9 @@ def update_nodegroup_status(self, context, cluster, nodegroup):
                 nodegroup.status = f"{action}_COMPLETE"
             elif phase in ("Failed", "Unknown"):
                 nodegroup.status = f"{action}_FAILED"
-
+            # Note: Fetch node_count from md object when autoscaling enabled
+            if utils.get_auto_scaling_enabled(cluster):
+                nodegroup.node_count = md.obj["spec"]["replicas"]
```

Comment:
The issue is that this will only be called if the cluster status is `*_IN_PROGRESS` which means it will be out of sync.

I think the better place is to move this to the health status updates

Reply:
yeah, i already found where i have to poll.
I never knew we have our own `monitor` in the driver already